src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -185,15 +186,19 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
         NotificationMessage notificationMessage = null;
         for (int i = 1; i <= selections.length; i++) {
             String selection = selections[i - 1];
-            boolean gotNextScreen = menuSession.handleInput(selection);
+
+            // minimal entity screens are only safe if there will be no further selection
```

Comment:
I understand why we would need to keep the screen for a case detail, but what accounts for the 'no further selection' limitation?

Reply:
To the extent I can tell, app navigation works by feeding selections as input to CLI screens (thats the loop this line is in). So for anything where a case is part of the input (like when entering a form), the `EntityScreen` needs to be able to handle case selection, and at least currently that requires the two components that I skipped in the minimal version.

Reply:
ah, I see. This is about the future requests that are 'playing back' the user choices that we talked about as one of the other potential targets for performance improvement. That does sound right. Thanks for explaining